### PR TITLE
Enable Umbrel systemd services at build

### PR DESCRIPTION
**Pull request by mayankchhabra**
This removes the need of having, updating and maintaining a boot script that's executed via `rc.local` and also decreases the first boot time.

Another benefit of not having a boot script is to not worry about where the `boot` script is executed from (external storage or SD card). The only thing we need to update now during OTA updates is installing the systemd services, and that's it. 

Also, MUDA!

Related: https://github.com/getumbrel/umbrel/pull/126

### CODE REVIEW

File: stage2/03-install-umbrel/01-run.sh [hunk 1]
```diff
@@ -22,36 +22,41 @@ EOF
...
+EOF
+done
+
+# Copy Umbrel to image
+mkdir "${ROOTFS_DIR}/home/${FIRST_USER_NAME}/umbrel"
+rsync -qavPHSX /umbrel "${ROOTFS_DIR}/home/${FIRST_USER_NAME}/"
+
+# Fix permissions
 on_chroot << EOF
-sed -i -e "s/\/home\/umbrel/\/home\/${FIRST_USER_NAME}/g" scripts/umbrel-os/umbrel-details
```
Comment (AaronDewes) [inline stage2/03-install-umbrel/01-run.sh:42]: This command is still needed. https://github.com/getumbrel/umbrel/blob/master/scripts/umbrel-os/umbrel-details#L84

Comment (mayankchhabra) [reply stage2/03-install-umbrel/01-run.sh:42]: Good find! Thanks

File: stage2/03-install-umbrel/01-run.sh [hunk 2]
```diff
@@ -22,36 +22,44 @@ EOF
...
+systemctl enable "${service}"
+EOF
+done
+
+# Replace /home/umbrel with home/$FIRST_USER_NAME in other scripts
+sed -i -e "s/\/home\/umbrel/\/home\/${FIRST_USER_NAME}/g" "/umbrel/scripts/umbrel-os/umbrel-details"
+
+# Copy Umbrel to image
+mkdir "${ROOTFS_DIR}/home/${FIRST_USER_NAME}/umbrel"
+rsync -qavPHSX /umbrel "${ROOTFS_DIR}/home/${FIRST_USER_NAME}/"
```
Comment (lukechilds) [inline stage2/03-install-umbrel/01-run.sh:52]: Would be good if we could comment what these `rsync` options mean.

Alternatively, if they have long alternatives e.g `-q` can be `--quiet`, we could use those instead as they're self documenting.

**Comment by lukechilds** [comment]
Looks good.

>The only thing we need to update now during OTA updates is installing the systemd services, and that's it.

I still think we should update the entire SD card install on OTA updates. For example if the user upgrades the SSD in the future we'll copy the SD card Umbrel install over to it during reformatting. If the services are out of sync with the Umbrel install that could cause issues.

File: stage2/03-install-umbrel/01-run.sh (hunk 2 above)
Comment (mayankchhabra) [reply stage2/03-install-umbrel/01-run.sh:52]: Good idea, will do

**Comment by mayankchhabra** [comment]
Oh, yes we should definitely upgrade the SD card version too. Sorry I wrongly worded this:

>  The only thing we need to update now during OTA updates is installing the systemd services, and that's it.

**Pull request merged**